### PR TITLE
feat(transport): websys-webtransport

### PR DESCRIPTION
**Pull request by mriise**
## Description
WIP Draft

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: transports/websys-webtransport/.cargo/config.toml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# WebTransport is a working draft! Remove when web-sys moves the interface out of unstable.
+[build]
+rustflags = ["--cfg=web_sys_unstable_apis"]
+rustdocflags = ["--cfg=web_sys_unstable_apis"]
```
Comment (mriise) [inline transports/websys-webtransport/.cargo/config.toml:4]: Note that these only applies when running `cargo build/doc` at this crate level.

File: transports/websys-webtransport/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,45 @@
+
+use web_sys::WebTransport;
+use libp2p_core::Transport;
+
+pub struct WebTransportTransport;
```
Comment (thomaseizinger) [inline transports/websys-webtransport/src/lib.rs]:
```suggestion
pub struct Transport;
```

See #2217 for the naming convention we are trying to follow.

File: transports/websys-webtransport/.cargo/config.toml (hunk 1 above)
Comment (thomaseizinger) [reply transports/websys-webtransport/.cargo/config.toml:4]: Would it make sense to add this to the docs.rs table in the `Cargo.toml`?

**Review by thomaseizinger — COMMENTED** [review]
Looking forward to this!

**Comment by thomaseizinger** [comment]
Closing this in favor of the almost ready https://github.com/libp2p/rust-libp2p/pull/4015.

Thanks for getting it started @mriise !

**Pull request closed without merging**